### PR TITLE
Publish compiled build on merge to develop

### PR DESCRIPTION
**Pull request by chris-fo**
## Description

This PR solves #147.

### Steps completed:

According to our definition of done, I have completed the following steps:

- [ ] User stories met
- [ ] Storybook stories
- [ ] Unit tests added
- [ ] Linting passing
- [ ] Unit tests passing
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

### CODE REVIEW

File: .github/workflows/release-package.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      packages: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          registry-url: https://npm.pkg.github.com/
+      - run: yarn
+      - run: yarn publish --out /artifacts/%s-%v.tgz
```
Comment (dbudzins) [inline .github/workflows/release-package.yml:26]: Take a look at this action: https://github.com/actions/upload-artifact

File: .github/workflows/release-package.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches: [ $default-branch ]
+    types: [closed]
+
+jobs:
+  # only publish artifact when PR was merged, not if it was closed without merging
+  publish-artifact:
+    if: github.event.pull_request.merged == true
+    runs-on: macos-latest
```
Comment (budzworthy) [inline .github/workflows/release-package.yml:12]: This should be ubuntu-latest

File: .github/workflows/release-package.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: Release Package
+
+on:
+  push:
+    branches: [ $default-branch ]
+    types: [closed]
```
Comment (budzworthy) [inline .github/workflows/release-package.yml:6]: What does type closed mean?  Pushes to the develop branch shouldn't have an open/closed status (we're triggering on the push itself not the PR)

File: .github/workflows/release-package.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  # only publish artifact when PR was merged, not if it was closed without merging
+  publish-artifact:
+    if: github.event.pull_request.merged == true
```
Comment (budzworthy) [inline .github/workflows/release-package.yml:11]: Again, this shouldn't factor in when the trigger is a push

**Comment by chris-fo** [comment]
Not needed

**Pull request closed without merging**